src/cli.js

```diff
@@ -105,13 +105,15 @@ async function* outputPrettyJSON(result) {
 }
 
 async function* outputNDJSON(result) {
-  if (result.getType() == 'array') {
-    for await (const value of result) {
-      yield JSON.stringify(await value.get())
+  const value = await result.get()
+
+  if (Array.isArray(value)) {
+    for await (const row of value) {
+      yield JSON.stringify(row)
       yield '\n'
     }
   } else {
-    yield JSON.stringify(await result.get())
+    yield JSON.stringify(value)
     yield '\n'
   }
 }
```

Comment:
I think all we want to do is change L108 to `if (result.type == 'stream') {` that way we stream the values per document in the resulting query

The proposed solution would turn any arrays into ndjson separated list, `echo '{\"foo\": [1,2,3,4]}\n{\"bar\": [2]}' | ./bin/groq.js -n --pretty '*[][0].foo'` would return '1\n2\n3\n4` instead of `[1,2,3,4]`